x/protorev/keeper/hooks_test.go

```diff
@@ -0,0 +1,775 @@
+			name: "Get Balancer Pool Comparable Liquidity",
+			param: param{
+				executePoolCreation: func() uint64 {
+					return s.PrepareBalancerPoolWithCoins(sdk.NewCoin("uosmo", sdk.NewInt(10)), sdk.NewCoin("juno", sdk.NewInt(10)))
+				},
+				expectedComparableLiquidity: sdk.NewInt(100),
+			},
+			expectPass: true,
+		},
+		{
+			name: "Get Stable Swap Pool Comparable Liquidity",
+			param: param{
+				executePoolCreation: func() uint64 {
+					return s.createStableswapPool(
+						sdk.NewCoins(
+							sdk.NewCoin("uosmo", sdk.NewInt(10)),
+							sdk.NewCoin("juno", sdk.NewInt(10)),
+						),
+						stableswap.PoolParams{
+							SwapFee: sdk.NewDecWithPrec(1, 4),
+							ExitFee: sdk.NewDecWithPrec(0, 2),
+						},
+						[]uint64{1, 1},
+					)
+				},
+				expectedComparableLiquidity: sdk.NewInt(100),
+			},
+			expectPass: true,
+		},
+		{
+			name: "Get Concentrated Liquidity Pool Comparable Liquidity",
+			param: param{
+				executePoolCreation: func() uint64 {
+					pool := s.PrepareConcentratedPool()
+					err := s.App.BankKeeper.SendCoins(
+						s.Ctx,
+						s.TestAccs[0],
+						pool.GetAddress(),
+						sdk.NewCoins(sdk.NewCoin("eth", sdk.NewInt(10)), sdk.NewCoin("usdc", sdk.NewInt(10))))
+					s.Require().NoError(err)
+					return pool.GetId()
+				},
+				expectedComparableLiquidity: sdk.NewInt(100),
+			},
+			expectPass: true,
+		},
+	}
+
+	for _, tc := range tests {
+		s.Run(tc.name, func() {
+			s.SetupTest()
+
+			// Create the pool
+			poolId := tc.param.executePoolCreation()
+
+			// Get the comparable liquidity
+			comparableLiquidity, err := s.App.ProtoRevKeeper.GetComparablePoolLiquidity(s.Ctx, poolId)
+
+			if tc.expectPass {
+				s.Require().NoError(err)
+				s.Require().Equal(tc.param.expectedComparableLiquidity, comparableLiquidity)
+			}
+		})
+	}
+}
+
+// Tests StoreJoinExitPoolSwaps stores the swaps associated with GAMM join/exit pool messages in the store, depending on if it is a join or exit.
+func (s *KeeperTestSuite) TestStoreJoinExitPoolSwaps() {
+	type param struct {
+		poolId       uint64
+		denom        string
+		isJoin       bool
+		expectedSwap types.Trade
+	}
+
+	tests := []struct {
+		name       string
+		param      param
+		expectPass bool
+	}{
+		{
+			name: "Store Join Pool Swap",
+			param: param{
+				poolId: 1,
+				denom:  "akash",
+				isJoin: true,
+				expectedSwap: types.Trade{
+					Pool:     1,
+					TokenIn:  "akash",
+					TokenOut: "Atom",
+				},
+			},
+			expectPass: true,
+		},
+		{
+			name: "Store Exit Pool Swap",
+			param: param{
+				poolId: 1,
+				denom:  "akash",
+				isJoin: false,
+				expectedSwap: types.Trade{
+					Pool:     1,
+					TokenIn:  "Atom",
+					TokenOut: "akash",
+				},
+			},
+			expectPass: true,
+		},
+	}
+
+	for _, tc := range tests {
+		s.Run(tc.name, func() {
+			s.SetupTest()
+
+			// All pools are already created in the setup
+			s.App.ProtoRevKeeper.StoreJoinExitPoolSwaps(s.Ctx, s.TestAccs[0], tc.param.poolId, tc.param.denom, tc.param.isJoin)
```

Comment:
can we have an error case for this where it fails because pool id is not a gamm pool?

Reply:
Added here: https://github.com/osmosis-labs/osmosis/pull/5045/commits/c2db6d772fad7e4fe77770e757e5561d05ac9c75